tempesta_fw/http.c

```diff
@@ -487,7 +487,7 @@ tfw_http_send_resp(TfwHttpReq *req, resp_code_t code)
 		msg.len += crlf->len - crlf_len;
 	}
 
-	if (!(hmresp = tfw_http_msg_alloc_err_resp()))
+	if (!(hmresp = tfw_http_msg_alloc_err_resp(req)))
```

Comment:
Isn't `resp->req` irrelevant at this stage, other that for a `BUG_ON()`?

At this stage the paired request is definitely not in the forwarding queue. From there, the response and the paired request go together through the code function call chain where necessary (as function arguments). This can be viewed as loose binding.

With that in mind, perhaps, only one of the two ways of binding should be used once the response is parsed. One is direct binding, where the link to the paired request is kept in the response's structure. This comes for free now. The other is loose binding where the paired request comes as an argument in function call chain in the code.

I don't know how easy or difficult would be to do that. But using both of these ways of binding seems to be a little confusing now (and, perhaps, excessive).

Reply:
Idea was to use `resp->req` linkage as a primary linkage. Response just can't exist if there is no request for it.  But the previous loose binding was kept untouched in the patch. Now i see how it's confusing, so I plan to keep only first one.

Reply:
Yes, please address the comment, e.g. `tfw_http_resp_fwd()` doesn't need to accept `req` as an argument at the below.

By the way, @keshonok thank you for the meaningful review.